lib/commands/gesture.js

```diff
@@ -327,11 +336,99 @@ helpers.mobileScroll = async function (opts={}, swipe=false) {
     log.errorAndThrow(msg);
   }
 
-  let element = opts.element.ELEMENT ? opts.element.ELEMENT : opts.element;
+  let element = opts.element.ELEMENT || opts.element;
   let endpoint = `/wda/element/${element}/${swipe ? 'swipe' : 'scroll'}`;
   return await this.proxyCommand(endpoint, 'POST', params);
 };
 
+function parseFloatParameter (paramName, paramValue, methodName) {
+  if (_.isUndefined(paramValue)) {
+    log.errorAndThrow(`"${paramName}" parameter is mandatory for "${methodName}" call`);
+  }
+  if (isNaN(paramValue)) {
+    log.errorAndThrow(`"${paramName}" parameter should be a valid number. "${paramValue}" is given instead`);
+  }
+  return parseFloat(paramValue);
+}
+
+helpers.mobilePinch = async function (opts={}) {
+  if (!opts.element) {
+    opts.element = await this.findNativeElementOrElements(`class name`, `XCUIElementTypeApplication`, false);
+  }
+  const params = {
+    scale: parseFloatParameter('scale', opts.scale, 'pinch'),
+    velocity: parseFloatParameter('velocity', opts.velocity, 'pinch')
+  };
+  const el = opts.element.ELEMENT || opts.element;
+  await this.proxyCommand(`/wda/element/${el}/pinch`, 'POST', params);
```

Comment:
awesome, however what i was subsequently reflecting on is that we'd need to have a `return` here, and in each other instance of `proxyCommand` inside a `mobile` method as well, if we want the outer `return` you just added to mean anything :-)